kernel/syscall.c

```diff
@@ -797,49 +797,108 @@ long myst_syscall_open(const char* pathname, int flags, mode_t mode)
     return ret;
 }
 
+// Caller should check *abspath_out before free:
+//
+//   if (*abspath_out != pathname)
+//        free(*abspath_out)
```

Comment:
Some comments here would help I think. Especially about the null string return.

Reply:
I will add some comments in another PR. Merged this PR first to unblock #542